charters/UploadService/charter.md

```diff
@@ -0,0 +1,78 @@
+# Upload Service
+
+## Description
+The Upload Service enables fast, cost-effective upload, staging, and validation of data submitted to the Human Cell Atlas (HCA) Data Coordination Platform (DCP) from across the world.
+
+## Objectives
+1. Enable submitters to upload files quickly and cost-effectively to temporary *staging areas* for subsequent submission to the Data Storage Service (DSS)
+1. Provide tools a command line interface (CLI) and APIs to upload files to staging areas
+1. Provide DCP developers the ability to validate files loaded into staging areas with user-defined validation jobs
+
+## In-scope
+* Provide staging areas that enable fast and cost-effective data transfer between:
+    * The DCP Ingestion UI workflow and the DSS
+    * The DCP Ingestion *data wrangler* workflow and the DSS
+    * The DCP Secondary Analysis Service and the DSS
+* Provide APIs for interacting with the upload service
+    * Functionality for other DCP modules (primarily the DCP Ingestion Infrastructure)
+        * Create and delete staging areas
+        * Schedule and retrieve validation jobs for files
+        * Store metadata content into staging area
+    * Functionality for Submitters
+        * Obtain credentials to write to upload areas
+        * Examine the contents of upload areas
+* Provide data integrity checks
+    * Compute checksums for all staged files and make these checksums available to the DSS
+    * Submitter must be able to cryptographically confirm that their data was transferred with integrity
+* Run user-defined validation jobs on files
+    * Provide environment to run containerized validation jobs on data files
+    * Provide job scheduling, execution, and status notifications for jobs
+* Provide tooling for submitters
+    * Command line tools for uploading data into the system and system introspection
+    * User authentication and authorization
+* Provide libraries and tooling for automating and inspecting upload jobs
+    * Event auditing for DCP operators enabling tracking job status
+    * CI/CD automation for easing development and deployment
+* Scale testing and optimization
+    * Build applications to scale elastically with size of workload
+    * Changes to the system to optimize cost and throughput
+    * Data routing optimization and configuration that increases data transfer rates
+    * Enable responsive workflow for data wranglers uploading datasets manually
+* User experience design
+    * Technical UX review with data wranglers and product iteration based on feedback
+* Dependencies
+    * User-defined jobs to validate files
```

Comment:
I think integration with the ingest lifecycle needs to be mentioned here; upload provides notifications about file upload, checksums and validation back to ingest to enable event tracking